base/parse.jl

```diff
@@ -167,6 +167,29 @@ float(x::AbstractString) = parse(Float64,x)
 
 float{S<:AbstractString}(a::AbstractArray{S}) = map!(float, similar(a,typeof(float(0))), a)
 
+function parse{T<:AbstractFloat}(::Type{T}, s::AbstractString, base::Integer)
```

Comment:
A lot of edge cases are missing:
negative numbers, NaN, Inf, trailing whitespace etc.